lib/src/features/user/ui/perf_stats_screen.dart

```diff
@@ -0,0 +1,457 @@
+            ]),
+            statGroupSpace,
+            _customPlatformCard(context, context.l10n.winningStreak, widgets: [
+              ..._streakWidget(context, data.maxWinStreak, data.curWinStreak,
+                  color: LichessColors.good),
+            ]),
+            _customPlatformCard(context, context.l10n.losingStreak, widgets: [
+              ..._streakWidget(context, data.maxLossStreak, data.curLossStreak,
+                  color: LichessColors.red),
+            ]),
+            _customPlatformCard(context, context.l10n.gamesInARow, widgets: [
+              ..._streakWidget(context, data.maxPlayStreak, data.curPlayStreak),
+            ]),
+            _customPlatformCard(context, context.l10n.maxTimePlaying, widgets: [
+              ..._streakWidget(context, data.maxTimeStreak, data.curTimeStreak,
+                  isTimeStreak: true),
+            ]),
+            statGroupSpace,
+            _customPlatformCard(context, context.l10n.bestRated,
+                widgets: _gameListWidget(context, data.bestWins)),
+            _customPlatformCard(context, context.l10n.worstRated,
+                widgets: _gameListWidget(context, data.worstLosses))
+          ]),
+        ));
+      },
+      error: (error, stackTrace) {
+        debugPrint(
+            'SEVERE: [PerfStatsScreen] could not load user games; ${error.toString()}\n$stackTrace');
+        return const Center(child: Text('Could not load user stats.'));
+      },
+      loading: () => const CenterLoadingIndicator(),
+    );
+  }
+
+  String _getPercentageString(num numerator, num denominator) {
+    return '${((numerator / denominator) * 100).round().toString()}%';
+  }
+
+  PlatformCard _customPlatformCard(
+    BuildContext context,
+    String stat, {
+    List<Widget>? widgets,
+    String? value,
+    TextStyle? styleStat,
+    TextStyle? styleValue,
+  }) {
+    final defaultStatStyle = TextStyle(
+      color: textShade(context, _customOpacity),
+      fontSize: _defaultStatFontSize,
+    );
+
+    const defaultValueStyle = TextStyle(fontSize: _defaultValueFontSize);
+
+    final TextStyle trueStatStyle = styleStat ?? defaultStatStyle;
+    final TextStyle trueValueStyle = styleValue ?? defaultValueStyle;
+
+    return PlatformCard(
+        child: Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.spaceAround,
+        children: [
+          FittedBox(
+              alignment: Alignment.center,
+              fit: BoxFit.scaleDown,
+              child: Text(stat,
+                  style: trueStatStyle, textAlign: TextAlign.center)),
+          if (value != null)
+            Text(value, style: trueValueStyle, textAlign: TextAlign.center)
+          else if (widgets != null)
+            ...widgets
+          else
+            Text('?', style: trueValueStyle)
+        ],
+      ),
+    ));
+  }
+
+  IntrinsicHeight _rowFromCards(List<PlatformCard> cards) {
```

Comment:
Same remark: make a private `class` instead of function (the remark applies to all other cases)